docs/developers-guide/substreams-sink-files.md

```diff
@@ -0,0 +1,230 @@
+    ]
+  }
+}
+```
+
+## Substreams modifications
+
+### Module handler changes for sink
+
+The example code in the [`lib.rs`](#) Rust source code file contains the `jsonl_out` module handler responsible for extracting the blockchain data. The module handler is responsible for passing the data to the `sf.substreams.sink.files.v1` protobuf for the sink tool and its processes.
+
+```rust
+#[substreams::handlers::map]
+fn jsonl_out(block: eth::Block) -> Result<Lines, substreams::errors::Error> {
+
+    let header = block.header.as_ref().unwrap();
+
+    Ok(pb::sinkfiles::Lines {
+        lines: vec![json!({
+            "number": block.number,
+            "hash": Hex(&block.hash).to_string(),
+            "parent_hash": Hex(&header.parent_hash).to_string(),
+            "timestamp": header.timestamp.as_ref().unwrap().to_string()
+        })
+        .to_string()],
+    })
+}
+```
+
+The module handler uses `JSONL` for the output type, `CSV` is also supported.. The [`json!`](https://docs.rs/serde_json/latest/serde_json/macro.json.html) macro is used to write the block data to the Rust `Vec` type by using the Rust [`vec!`](https://doc.rust-lang.org/std/macro.vec.html) macro.
+
+**TODO**: <i>We just need to emphasis in the tutorial that it's an example and that a lot of "rows" can be extracted per block. Giving the ERC20/ERC721/ERC1155 examples is a good idea as we coded it: https://github.com/streamingfast/substreams-eth-token-transfers/blob/develop/src/lib.rs#L24</i>
+
+**TODO:** <i>How to respect Sink's expected output's type with examples for JSON (maybe CSV))</i>
+
+## Understanding the sink tool
+
+### Run and configure the `substreams-sink-files` tool
+
+The command to start and run the `substreams-sink-files` tool for the accompanying Substreams project will resemble:
+
+{% code overflow="wrap" %}
+
+```bash
+substreams-sink-files run --encoder=lines --state-store=./localdata/working/state.yaml mainnet.eth.streamingfast.io:443 substreams.yaml jsonl_out ./localdata/out
+```
+
+{% endcode %}
+
+Flags
+
+- `file_output_path`
+  The path to a directory the sink will write its files to during processing.
+- `file_working_dir`
+  **TODO:** <i>Description here.</i>
+- `endpoint`
+  The URI of the firehose service Substreams is connecting to.
+- `encoder`
+  **TODO:** <i>Description here.</i>
+- `manifest_path`
+  The path to the Substreams manifest for the Substreams module prepared for the sink.
+- `output_module_name`
+  The name of the Rust-based Substreams module the sink will run during its processing.
+- `block_range`
+  **TODO:** <i>Description here.</i>
+- `state_store`
+  **TODO:** <i>Description here.</i>
+- `blocks_per_file`
+  The number of Block objects from the blockchain to process while the sink is processing.
+- `buffer_max_size`
+  **TODO:** <i>Description here.</i>
+
+**TODO:**
+<i>
```

Comment:
We won't copy over each flag's description. Links to GitHub source code for flag definition will be enough. I'll open an issue to write the flags description correctly.

I can see that we ask people to use `substreams-sink-files run --help` to see description of each flag and we quickly link to the online defintion https://github.com/streamingfast/substreams-sink-files/blob/master/cmd/substreams-sink-files/run.go#L31 using a quick markdown link so people could go there and see directly.

Reply:
Ok, sounds good. There were instructions to include flag information in the original request for this documentation. The flags have now been removed.